src/Network/GRPC/Common/HTTP2Settings.hs

```diff
@@ -0,0 +1,95 @@
+-- | Settings and parameters pertaining to HTTP\/2
+--
+-- Intended for unqualified import.
+
+module Network.GRPC.Common.HTTP2Settings
+  ( HTTP2Settings(..)
+  , defaultHTTP2Settings
+  ) where
+
+import Data.Default
+import Data.Word
+
+-- | HTTP\/2 settings
+data HTTP2Settings = HTTP2Settings {
```

Comment:
I like that you bundled all of these parameters into a separate record, feels nice and clean.